segmentation_models_pytorch/pan/model.py

```diff
@@ -0,0 +1,87 @@
+from typing import Optional, Union
+from .decoder import PANDecoder
+from ..encoders import get_encoder
+from ..base import SegmentationModel
+from ..base import SegmentationHead, ClassificationHead
+
+
+class PAN(SegmentationModel):
+    """ Implementation of _PAN (Pyramid Attention Network)
+
+    Args:
+        encoder_name: name of classification model (without last dense layers) used as feature
+            extractor to build segmentation model.
+        encoder_depth (int): number of stages used in decoder, larger depth - more features are generated.
+            e.g. for depth=3 encoder will generate list of features with following spatial shapes
+            [(H,W), (H/2, W/2), (H/4, W/4), (H/8, W/8)], so in general the deepest feature tensor will have
+            spatial resolution (H/(2^depth), W/(2^depth)]
+        encoder_weights: one of ``None`` (random initialization), ``imagenet`` (pre-training on ImageNet).
+        decoder_channels: Number of ``Conv2D`` layer filters in decoder blocks
+        gau_upscale_mode: Upscale mode for decoder;
+            One of [``nearest``, ``'inear``, ``bilinear``, ``bicubic``, ``trilinear``, ``area``]. Default: ``bilinear``
+        in_channels: number of input channels for model, default is 3.
+        classes: a number of classes for output (output shape - ``(batch, classes, h, w)``).
+        activation: activation function to apply after final convolution;
+            One of [``sigmoid``, ``softmax``, ``logsoftmax``, ``identity``, callable, None]
+        upsampling: optional, final upsampling factor
+            (default is 4 to preserve input -> output spatial shape identity)
+        aux_params: if specified model will have additional classification auxiliary output
+            build on top of encoder, supported params:
+                - classes (int): number of classes
+                - pooling (str): one of 'max', 'avg'. Default is 'avg'.
+                - dropout (float): dropout factor in [0, 1)
+                - activation (str): activation function to apply "sigmoid"/"softmax" (could be None to return logits)
+
+    Returns:
+        ``torch.nn.Module``: **PAN**
+
+    .. _PAN:
+        https://arxiv.org/abs/1805.10180
+
+    """
+
+    def __init__(
+            self,
+            encoder_name: str = "resnet34",
+            encoder_depth: int = 5,
+            encoder_weights: str = "imagenet",
+            decoder_channels: int = 32,
+            gau_upscale_mode: str = 'bilinear',
```

Comment:
Suggestion: rename -> `decoder_upsampling_mode` (just here for models API unification)

Reply:
Do we actually need to change this parameter? What about hardcoded "bilinear"?